src/table/utils/styling-utils.js

```diff
@@ -74,27 +105,32 @@ export function getBodyStyle(layout, theme) {
   // the same as whatever the font color is when the hover background
   // color is a light color.
   // 2. There is a hover font color but no hover background color,
-  // when hovering, only a hover font color is applied and the hover
-  // background color disappears.
+  // when hovering, only the font color is applied and no hover
+  // background color is shown.
   // 3. There is no hover font color and no hover background color,
-  // when hovering, the default hover effect (light gray background)
-  // 4. There are both hover font and background colors, when hovering,
-  // the hover font and background colors take effect.
-
-  const unsetHoverBackgroundColor = isUnset(content?.hoverColor);
-  const unsetHoverFontBackgroundColor = isUnset(content?.hoverFontColor) && unsetHoverBackgroundColor;
-
-  const setHoverBackgroundColor = unsetHoverBackgroundColor
-    ? ''
-    : getColor(STYLING_DEFAULTS.HOVER_BACKGROUND, theme, content?.hoverColor);
-
-  const hoverBackgroundColor = unsetHoverFontBackgroundColor
-    ? STYLING_DEFAULTS.HOVER_BACKGROUND
-    : setHoverBackgroundColor;
+  // when hovering, the default hover effect (a light gray hover background
+  // color and no hover font color).
+  // 4. There are both a hover font and a hover background color,
+  // when hovering, the hover font and the hover background color take effect.
+  // 5. The hover font color and hover background color from properties have a
+  // higher priorities than those from theme.
```

Comment:
This isn't really a 5th case, 
```suggestion
  // Note that hover colors from properties have a
  // higher priority than those from theme.
```